core/shared/src/main/scala/fs2/Stream.scala

```diff
@@ -642,33 +639,32 @@ final class Stream[+F[_], +O] private[fs2] (private[fs2] val underlying: Pull[F,
     */
   def debounce[F2[x] >: F[x]](
       d: FiniteDuration
-  )(implicit F: Temporal[F2]): Stream[F2, O] =
-    Stream.eval(Queue.bounded[F2, Option[O]](1)).flatMap { queue =>
-      Stream.eval(F.ref[Option[O]](None)).flatMap { ref =>
-        val enqueueLatest: F2[Unit] =
-          ref.getAndSet(None).flatMap {
-            case v @ Some(_) => queue.offer(v)
-            case None        => F.unit
-          }
+  )(implicit F: Temporal[F2]): Stream[F2, O] = Stream.force {
+    for {
+      queue <- Queue.bounded[F2, Option[O]](1)
+      ref <- F.ref[Option[O]](None)
+    } yield {
+      val enqueueLatest: F2[Unit] =
+        ref.getAndSet(None).flatMap(prev => if (prev.isEmpty) F.unit else queue.offer(prev))
```

Comment:
Is this... `prev.traverse_(queue.offer)`?

Reply:
Yes.

Reply:
`flatMap(_.traverse_(queue.offer))`

Reply:
I had noticed that it is `traverse_` and had looked at it... but then I decanted towards explicitly using `prev` to indicate it is the previous value, if any, that is inserted to queue.